src/main/java/sparksoniq/jsoniq/runtime/iterator/RuntimeIterator.java

```diff
@@ -45,12 +45,15 @@
 
```

Comment:
That's what I missed -- it is actually serialized with kryo, please forget my last question in the email.

Reply:
I tried setting breakpoints for the write and read methods of RuntimeIterator but nothing was hit when i tried a few queries manually and our regular tests as well. I also tried to falsify the values in the read method (so that isOpen and hasNext would be set to true upon object instantiation) but code ran normally again.

When I tried to set similar breakpoints for write and read methods of BooleanItem, the breakpoints were actually hit in some of the tests so I believe that the kryo serialization for RuntimeIterators are in fact not used.

Please let me know if you have a better testing method in mind or if there is something missing in my understanding.

Reply:
A place to look at would be the Spark configuration. There are two things that should be configurable: the serialization of results (the Items in the RDDs), which Stefan made use Kryo already during his thesis (plus you did the same for manual serialization in DataFrames), and on the other hand the serialization of all closures passed to Spark in transformations (and this is were the Iterators are serialized), which we are unsure uses Java IO or Kryo.

While I do not have the exact parameters on my mind, this may be a useful reading to start with: 

https://ogirardot.wordpress.com/2015/01/09/changing-sparks-default-java-serialization-to-kryo/

However, since for now the matter with FunctionItems is fixed with using Java IO for the iterator, there is no urgency in this. You can look into this when you see fit.